src/multisig.js

```diff
@@ -0,0 +1,184 @@
+            txn.name = "Transaction";
+            txn.tag = Buffer.from([84, 88]); // "TX"
+
+            txn.objForEncoding = txnForEnc;
+            return txn;
+        }
+        return super.from_obj_for_encoding(txnForEnc);
+    }
+
+    /**
+     * partialSignTxn partially signs this transaction and returns a partially-signed multisig transaction,
+     * encoded with msgpack as a typed array.
+     * @param version multisig version
+     * @param threshold multisig threshold
+     * @param pks multisig public key list, order is important.
+     * @param sk an Algorand secret key to sign with.
+     * @returns an encoded, partially signed multisig transaction.
+     */
+    partialSignTxn({version, threshold, pks}, sk) {
+        // verify one more time that the from field is correct
+        if (!this.hasOwnProperty("objForEncoding")) {
+            let expectedFromRaw = address.fromMultisigPreImg({version, threshold, pks});
+            if (address.encode(this.from.publicKey) !== address.encode(expectedFromRaw)) {
+                throw ERROR_MULTISIG_BAD_FROM_FIELD;
+            }
+        }
+        // get signature verifier
+        let myPk = nacl.keyPairFromSecretKey(sk).publicKey;
+        return createMultisigTransaction(
+            this.get_obj_for_encoding(),
+            {"rawSig": this.rawSignTxn(sk), myPk},
+            {version, threshold, pks},
+        );
+    }
+}
+
+/**
+ * createMultisigTransaction creates a multisig transaction blob.
+ * @param txnForEncoding the actual transaction to sign.
+ * @param rawSig a Buffer raw signature of that transaction
+ * @param myPk a public key that corresponds with rawSig
+ * @param version multisig version
+ * @param threshold mutlisig threshold
+ * @param pks ordered list of public keys in this multisig
+ * @returns encoded multisig blob
+ */
+function createMultisigTransaction(txnForEncoding, {rawSig, myPk}, {version, threshold, pks}) {
+    let keyExist = false;
+    // construct the appendable multisigned transaction format
+    let subsigs = pks.map(pk => {
+        if (nacl.bytesEqual(pk, myPk)) {
+            keyExist = true;
+            return {
+                "pk": Buffer.from(pk),
+                "s" : rawSig,
+            };
+        }
+        return {"pk": Buffer.from(pk)};
+    });
+    if (keyExist === false) {
+        throw ERROR_MULTISIG_KEY_NOT_EXIST;
+    }
+    let msig = {
+        "v": version,
+        "thr": threshold,
+        "subsig": subsigs,
+    };
+    let sTxn = {
+        "msig": msig,
+        "txn": txnForEncoding,
+    };
+    return new Uint8Array(encoding.encode(sTxn));
+}
+
+/**
+ * mergeMultisigTransactions takes a list of multisig transaction blobs, and merges them.
+ * @param multisigTxnBlobs a list of blobs representing encoded multisig txns
+ * @returns typed array msg-pack encoded multisig txn
+ */
+function mergeMultisigTransactions(multisigTxnBlobs) {
+    if (multisigTxnBlobs.length < 2) {
```

Comment:
same as above. Also, seems like you don't enforce the creation of threshold > 1

Reply:
I think its fairly reasonable two require two transactions in order to merge them.

Reply:
if the structure allows you to send one by itself, it's fine.